### DISCUSSION TITLE
Pi and Router has different internet IP

### DISCUSSION THREAD

**Discussion by Filston37** [pivpn/pivpn#1789] [Support]
I got pivpn and Pihole installed on a rasperry pi 3, it used to work like a charm, but after I updated to 1gb/s internet with a new router I can't get it to work anymore. The router brand is the same and the setup is identical to what I had previously.

I have gone though the FAQ on pivpn's webs site and I think I know why it does not work, but not how to fix it.

From the FAQ:

"

If you are already using a DDNS, and thus

pivpnHOST

contains your domain name, use

dig +short yourdomain.example.com

to check whether the returned IP matches

curl -s https://checkip.amazonaws.com

"

By runing the 2 commands I get 2 different IP addresses. The one returned by no-ip (my DDNS provider?) is the same reported by my router.

Why is my raspberry pi internet IP different than my router? and how can I change it?

Additional info:

Pi is on wifi, router is Linksys Velop Tri-Band AX5300 Mesh WiFi 6 Router, I don't have DHCP activated in the pi-hole but in the router.

**Comment by orazioedoardo** [comment]
Does the No-IP or router address fall in one of these ranges?

- 10.0.0.0 - 10.255.255.255
- 172.16.0.0 - 172.31.255.255
- 192.168.0.0 - 192.168.255.255
- 100.64.0.0 - 100.127.255.255

**Reply by Filston37** [reply]
yes, its 100.64.##.## both on router and noip.

**Reply by orazioedoardo** [reply]
Well, as I suspected your router is now under CGNAT, you need to ask the ISP for a public IP otherwise you can't actually open a port.

**Reply by Filston37** [reply]
Thanks, after reading up on the subject I think I understand the problem. My ISP does not offer static IPs is there a work around in pivpn or do I need to look for an other option to remote into my home?

**Reply by orazioedoardo** [reply]
The issue is not that the router IP is not static, you can use dynamic DNS to overcome that, but rather that your router gets assigned a private IP (100.64.X.Y), while the public IP is assigned to an ISP router inside their network and shared among many customers. Since you can't get access to their router, you can't port forward.

Solution is to either ask them to provide your router a public IP (dynamic is fine too), or have them configure a port forwarding. If they don't know any of this, likely they don't offer public IPs for consumers, but they might if a business plan is available.

As an alternative, you could install PiVPN on a cloud server or another Raspberry Pi in a network where you have a public IP and forward traffic to your home Raspberry Pi, but that's not a trivial task. Tailscale is a managed service that works around CGNAT, works quite well but it relies on you trusting one more software and service.
